static/src/javascripts/projects/common/modules/discussion/discussion-frontend.js

```diff
@@ -49,6 +56,20 @@ define([
         });
     }
 
+    function otherBannersVisible () {
+        var siteMessage = document.getElementsByClassName('js-site-message');
+        if (siteMessage.length && !siteMessage[0].classList.contains('is-hidden')) {
+            // Contribution banner is visible
+            return true;
+        }
+        var breakingNews = document.getElementsByClassName('js-breaking-news-placeholder');
+        if (breakingNews.length && !breakingNews[0].classList.contains('breaking-news--hidden')) {
+            // Breaking news alert is visible
+            return true;
+        }
+        return false;
+    }
+
```

Comment:
The plot thickens

`banner-message:complete` is emitted (or supposed to be emitted) in two places:

1. `donot-use-adblock`
2. `membership-engagement-banner`

However,  (1) is loaded by `enhanced/main` before asynchronously requiring `enhanced/trail`. So when I initialize `discussion-frontend` I've missed already the first `emit`.
(2) is only fired if you're in one of the variants of an AB test. Grantend that the test has 100% audience, but if you're `notintest` the event is not fired.

So listening to two events that may or may not be fired seems like a very brittle solution.
The solution maybe should be to refactor that logic and store in `ui/message` information about pending scripts that are trying to use the slot.

Another approach is to check there are no banners every time I try to render one. The performance cost is higher and given the previous consideration it seems unnecessary.

previous consideration: Discussion frontend is loaded async by `trail` which is loaded async by `enhanced` and ask (p)react to render a banner after making an API request. Most likely it'll be the last in the queue.

It's a short lived AB test (my last famous word), so if you don't mind I'd merge it like this.